include/dali/core/cuda_utils.h

```diff
@@ -21,8 +21,10 @@
 #include "dali/core/host_dev.h"
 #include "dali/core/dynlink_cuda.h"
 #include "dali/core/cuda_error.h"
+#ifndef __CUDA_ARCH__
+#include "dali/util/half.hpp"
+#endif
```

Comment:
Do we need the forward declaration, if we include it directly?

Reply:
I don't think so.